### PR TITLE
Add authentication example

### PR DESCRIPTION
**Pull request by SerayaEryn**
This adds a little authentication example.

### CODE REVIEW

**Review by jsumners — APPROVED** [review]
LGTM

File: fastify-session-authentication/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+This project can simply run by:
+
+```sh
+cd fastify-session-authentication
+npm install
+npm start
+```
+
+Then open `http://localhost:3000/login` in your browser.
```
Comment (Eomm) [inline fastify-session-authentication/README.md]:
```suggestion
Then open `http://localhost:3000/login` in your browser and use the `abcdef` password to login.
```

**Review by mcollina — CHANGES_REQUESTED** [review]
Would you mind adding some tests?

**Comment by SerayaEryn** [comment]
I added some tests.

File: fastify-session-authentication/authentication.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+const fastify = require('fastify')
+const fastifySession = require('fastify-session')
+const fastifyCookie = require('fastify-cookie')
+const fastifyFormbody = require('fastify-formbody')
+
+const { loginPage, defaultPage } = require('./html')
+
+// create a fastify instance
+const app = fastify()
```
Comment (mcollina) [inline fastify-session-authentication/authentication.js]: I would prefer if we exposed a factory that returned `app` without calling `app.listen()` and then call that only if this is the startup file.

File: fastify-session-authentication/authentication.test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+const { test } = require('tap')
+const server = require('./authentication')
+const requestCallback = require('request')
+
+server.then((app) => {
+  app.server.unref()
```
Comment (mcollina) [inline fastify-session-authentication/authentication.test.js]: We should instantiate one server for each test, ideally in a `beforeEach`.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: fastify-session-authentication/authentication.js (hunk 2 above)
Comment (Eomm) [reply fastify-session-authentication/authentication.js]: I would suggest using this approach with fastify-cli
https://github.com/fastify/example/blob/aa633714867da5edc029700145a5a7505be2b880/validation-messages/package.json#L7

so you write less code and more testable

Comment (SerayaEryn) [reply fastify-session-authentication/authentication.js]: I changed it to export a plugin and using `fastify-cli` to start the example.

File: fastify-session-authentication/authentication.test.js (hunk 3 above)
Comment (SerayaEryn) [reply fastify-session-authentication/authentication.test.js]: I instantiate one server for each test now.

**Comment by SerayaEryn** [comment]
I made changes to the example as suggested.

**Comment by L2jLiga** [comment]
@mcollina, friendly ping

**Review by mcollina — APPROVED** [review]
LGTM

**Pull request merged**